File: python-selenium-pytest-bdd/src/saucedemo_uitests/testdata/invalid_credentials_factory.py
"""Generates random credentials for negative login scenarios, guaranteed not to collide with a real
`user_catalog` account."""

from faker import Faker

from saucedemo_uitests.users.user_catalog import ALL_USERS


def create_invalid_credentials() -> tuple[str, str]:
    fake = Faker()

    username = fake.user_name()
    while username in ALL_USERS:
        username = fake.user_name()

    password = fake.password(length=12)
    return username, password

File: ts-playwright-test/src/testdata/invalidCredentialsFactory.ts
import { faker } from '@faker-js/faker';
import { allUsers } from '../users/userCatalog';

export interface InvalidCredentials {
  readonly username: string;
  readonly password: string;
}

/** Generates random credentials for negative login tests, guaranteed not to collide with a real userCatalog account. */
export function createInvalidCredentials(): InvalidCredentials {
  let username: string;
  do {
    username = faker.internet.username();
  } while (allUsers.has(username));

  return { username, password: faker.internet.password({ length: 12 }) };
}
